include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -6675,11 +6675,15 @@ namespace VectorTools
       // check that off-processor entries are zero. Otherwise we will compute
       // wrong results below!
       typename InVector::size_type i = 0;
-      typename Triangulation<dim,spacedim>::cell_iterator it = tria.begin_active();
-      for (; i<cellwise_error.size(); ++i, ++it)
-        if (!it->is_locally_owned())
-          Assert(std::fabs(cellwise_error[i]) <  1e-20,
-                 ExcMessage("cellwise_error of cells that are not locally owned need to be zero!"));
+      typename Triangulation<dim,spacedim>::active_cell_iterator it = tria.begin_active();
+      for (; i<cellwise_error.size(); ++i)
+        {
+          if (it == tria.end() || !it->is_locally_owned())
+            Assert(std::fabs(cellwise_error[i]) <  1e-20,
```

Comment:
In other words, I think the original way you wrote the loop was correct, you just needed to change the type of `it` to *active*.